tests/registry-e2e.rs

```diff
@@ -46,6 +47,40 @@ async fn expired_registrations_are_pruned_after_ttl() -> Result<(), anyhow::Erro
     Ok(())
 }
 
+#[tokio::test]
+async fn when_provider_registers_notifies_registry_observers() -> anyhow::Result<()> {
+    fn namespace_event(namespace: &str) -> String {
+        format!("namespaces/{}", namespace)
+    }
+
+    // arrange
+    let namespace = format!("e2e.registration.{}", get_uuid());
+
+    let builder = RegistrationBuilder::new(
+        "registration.provider.e2e",
+        "1.0.0",
+        "http://test-url:7090".parse().unwrap(), // arbitrary url, the provider will never be invoked
```

Comment:
nit: Why not `localhost` instead of `test-url`?

Reply:
Fixed this with a20a5330cf913881675d6eb0146c85d9e62a24b8.